Pomodoro.Api/Controllers/TasksController.cs

```diff
@@ -0,0 +1,186 @@
+        }
+
+        /// <summary>
+        /// Gets task by specific id.
+        /// </summary>
+        /// <param name="id">The day for which statistics should be returned.</param>
+        /// <returns>A <see cref="TaskModel"/> object.</returns>
+        [HttpGet("{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        [SwaggerResponse(200, "The task was found")]
+        [SwaggerResponse(400, "Invalid data")]
+        [SwaggerResponse(404, "The task with this id wasn`t found")]
+        [SwaggerResponse(500, "Something went wrong")]
+        public async Task<ActionResult<TaskModel>> GetTaskById(Guid id)
+        {
+            if (id == Guid.Empty)
+            {
+                return this.BadRequest();
+            }
+
+            var task = await this.tasksService.GetTaskByIdAsync(id);
+            if (task == null)
+            {
+                return this.NotFound();
+            }
+
+            return this.Ok(task);
+        }
+
+        /// <summary>
+        /// Gets all tasks by user.
+        /// </summary>
+        /// <returns>A <see cref="TaskModel"/> object.</returns>
+        [HttpGet("allUserTasks")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        [SwaggerResponse(200, "The tasks were found")]
+        [SwaggerResponse(401, "An unauthorized request cannot be processed.")]
+        [SwaggerResponse(500, "Something went wrong")]
+        public async Task<ActionResult<IReadOnlyList<TaskModel>>> GetTasksByUser()
+        {
+            var tasks = await this.tasksService.GetAllTasksAsync(this.UserId);
+            return this.Ok(tasks);
+        }
+
+        /// <summary>
+        /// Gets all tasks. The endpoint for test purposes.
+        /// </summary>
+        /// <returns>A <see cref="TaskModel"/> object.</returns>
+        [HttpGet("allTasksTest")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        [SwaggerResponse(200, "The task was found")]
+        [SwaggerResponse(500, "Something went wrong")]
+        public async Task<ActionResult<IReadOnlyList<TaskModel>>> GetAllTasksTest()
+        {
+            var tasks = await this.tasksService.GetAllTasksAsyncTest();
+            return this.Ok(tasks);
+        }
+
+        /// <summary>
+        /// Endpoint to create a task.
+        /// </summary>
+        /// <param name="task">Gets as a parameter Task create view model.</param>
+        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+        [HttpPost("postTask")]
```

Comment:
I can change it, but i don`t know, for me the routes can be long when they are descriptive. Also, we don`t pass the id when we post the task and pass it when we need to change or delete it. That's why there is a question: should we pass the id in the URL or shouldn't?

Reply:
Yes, we have to pass ids, because they are extremely necessary. I was talking about such cases when you duplicate names of http methods in your routes. It makes routes dirty, as for me.